pkg/prometheus/operator.go

```diff
@@ -1948,6 +1948,7 @@ func (c *Operator) selectProbes(ctx context.Context, p *monitoringv1.Prometheus,
 	var rejected int
 	res := make(map[string]*monitoringv1.Probe, len(probes))
 	for probeName, probe := range probes {
+
```

Comment:
suggestion: you can use an anonymous function to reduce code duplication

```golang
rejectFn := func(probe string, err error) {
        reject++
	level.Warn(c.logger).Log(
		"msg", "skipping probe",
		"error", err.Error(),
		"probe", probe,
		"namespace", p.Namespace,
		"prometheus", p.Name,
	)
}

for probeName, probe := range probes {
   if ... {
       rejectFn(probeName, err)
       continue
   }

   if ... {
       rejectFn(probeName, err)
       continue
   }

}
```

Reply:
Good idea, i have optimize the code.